sabelotodo/tests/test_routes.py

```diff
@@ -25,63 +28,140 @@ def test_item_route_with_empty_db(test_client, _db):
 def test_item_route_with_multiple_items(test_client, _db):
     """ The /item route retrieves all items added. """
 
-    i = Item(name="a", order=0, done=False)
-    j = Item(name="b", order=2, done=True)
-    k = Item(name="c", order=1, done=False)
+    items = [Item(name="a", order=0, done=False),
+             Item(name="b", order=2, done=True),
+             Item(name="c", order=1, done=False)]
 
-    _db.session.add(i)
-    _db.session.add(j)
-    _db.session.add(k)
+    _db.session.add_all(items)
     _db.session.commit()
 
     return_value = test_client.get('/item')
     assert json.loads(return_value.data) == [asdict(i) for i in Item.query.all()]
 
 
-def test_itemid_route_with_valid_id(test_client, _db):
+@pytest.mark.parametrize("idx", [0, 1, 2])
+def test_get_itemid_route_with_valid_id(test_client, _db, idx):
     """ The /item/<id> route retrieves just the item with the specified ID. """
 
-    i = Item(name="a", order=0, done=False)
-    j = Item(name="b", order=2, done=True)
-    k = Item(name="c", order=1, done=False)
+    items = [Item(name="a", order=0, done=False),
+             Item(name="b", order=2, done=True),
+             Item(name="c", order=1, done=False)]
 
-    _db.session.add(i)
-    _db.session.add(j)
-    _db.session.add(k)
+    _db.session.add_all(items)
     _db.session.commit()
 
-    this_id = j.id
-    return_value = test_client.get('/item/%s' % this_id)
-    assert json.loads(return_value.data) == asdict(j)
+    selection = items[idx]
+    return_value = test_client.get('/item/%s' % selection.id)
+    assert json.loads(return_value.data) == asdict(selection)
+
+
+@pytest.mark.parametrize("idx", [0, 1, 2])
+def test_delete_itemid_route_with_valid_id(test_client, _db, idx):
+    """ A DELETE request to /item/<id> deletes the item with the
+    specified ID. """
+
+    items = [Item(name="a", order=0, done=False),
+             Item(name="b", order=2, done=True),
+             Item(name="c", order=1, done=False)]
 
+    _db.session.add_all(items)
+    _db.session.commit()
 
-def test_itemid_route_failures(test_client, _db):
+    selection = items[idx]
+    remainder = items[:idx] + items[idx+1:]
+    return_value = test_client.delete('/item/%s' % selection.id)
+    assert return_value.status_code == 200
+    assert sorted(Item.query.all()) == sorted(remainder)
```

Comment:
I think I would also test that attempting to get the deleted item throws an error.